### ISSUE TITLE
Issues with husky command

### ISSUE THREAD

**Issue by Jairanpo** [kentcdodds/react-fundamentals#190]
# Error with node setup: 
Running the following command: npx "https://gist.github.com/kentcdodds/abbc32701f78fa70298d444c2303b6d9"
    �  Failure: System Validation. Please review the messages above for information on how to troubleshoot and resolve this issue.

## This is the tail of the traceback: 
6667 timing build:link:node_modules/webpack/node_modules/rimraf Completed in 349ms
6668 timing build:link Completed in 357ms
6669 info run husky@4.3.8 install node_modules/husky node husky install
6670 info run husky@4.3.8 install { code: 1, signal: null }
6671 timing reify:rollback:createSparse Completed in 15876ms
6672 timing reify:rollback:retireShallow Completed in 2237ms
6673 timing command:install Completed in 66915ms
6674 verbose stack Error: command failed
6674 verbose stack     at ChildProcess.<anonymous> (C:\Users\jaira\AppData\Roaming\npm\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:64:27)
6674 verbose stack     at ChildProcess.emit (events.js:315:20)
6674 verbose stack     at maybeClose (internal/child_process.js:1051:16)
6674 verbose stack     at Socket.<anonymous> (internal/child_process.js:442:11)
6674 verbose stack     at Socket.emit (events.js:315:20)
6674 verbose stack     at Pipe.<anonymous> (net.js:670:12)
6675 verbose pkgid husky@4.3.8
6676 verbose cwd D:\development\epicreact\react-fundamentals
6677 verbose Windows_NT 10.0.19043
6678 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\jaira\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install"
6679 verbose node v14.4.0
6680 verbose npm  v7.19.1
6681 error code 1
6682 error path D:\development\epicreact\react-fundamentals\node_modules\husky
6683 error command failed
6684 error command C:\Windows\system32\cmd.exe /d /s /c node husky install
6685 error D:\development\epicreact\react-fundamentals\node_modules\please-upgrade-node\index.js:7
6685 error   if (semverCompare(currentVersion, requiredVersion) === -1) {
6685 error       ^
6685 error
6685 error TypeError: semverCompare is not a function
6685 error     at pleaseUpgradeNode (D:\development\epicreact\react-fundamentals\node_modules\please-upgrade-node\index.js:7:7)
6685 error     at Object.<anonymous> (D:\development\epicreact\react-fundamentals\node_modules\husky\husky.js:7:1)
6685 error     at Module._compile (internal/modules/cjs/loader.js:1200:30)
6685 error     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
6685 error     at Module.load (internal/modules/cjs/loader.js:1049:32)
6685 error     at Function.Module._load (internal/modules/cjs/loader.js:937:14)
6685 error     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:71:12)
6685 error     at internal/main/run_main_module.js:17:47
6686 verbose exit 1

**Comment by kentcdodds** [comment]
Interesting! I'm guessing something went wrong with the installation of dependencies on your machine. Could you start over from scratch and see if it works the second time? If not, you might try one of the alternative ways to run the workshop (mentioned at the end of [the setup instructions](https://github.com/kentcdodds/react-fundamentals#setup))

**Comment by Jairanpo** [comment]
Hi Kent, thanks for the response!... So far I've been unlucky this is my result from docker: 

D:\development\epicreact\react-fundamentals>docker-compose up
Creating network "react-fundamentals_default" with the default driver
Building node
[+] Building 146.3s (9/9) FINISHED
 => [internal] load build definition from Dockerfile                0.3s
 => => transferring dockerfile: 133B                                0.1s
 => [internal] load .dockerignore                                   0.2s
 => => transferring context: 2B                                     0.0s
 => [internal] load metadata for docker.io/library/node:16          2.1s
 => [auth] library/node:pull token for registry-1.docker.io         0.0s
 => [1/4] FROM docker.io/library/node:16@sha256:ca6daf1543242acb0  48.3s
 => => resolve docker.io/library/node:16@sha256:ca6daf1543242acb0c  0.1s
 => => sha256:ca6daf1543242acb0ca59ff425509eab7def 1.21kB / 1.21kB  0.0s
 => => sha256:36df0d14549023f2191bd1ef11ed22ea80f0 2.21kB / 2.21kB  0.0s
 => => sha256:7105279fa2ab705ec483680834ebbc8c0570 7.80kB / 7.80kB  0.0s
 => => sha256:0bc3020d05f1e08b41f1c5d54650a157b 50.44MB / 50.44MB  12.7s
 => => sha256:a110e58716600c199fc95f633b30735c33a2 7.83MB / 7.83MB  2.3s
 => => sha256:83d3c0fa203acbade733bff627daa75b84 10.00MB / 10.00MB  2.8s
 => => sha256:a8fd09c11b021b756b7a92a4f70a3d444 51.84MB / 51.84MB  15.4s
 => => sha256:14feb89c4a52c44f0a7e23a46ebb76e 192.39MB / 192.39MB  35.6s
 => => extracting sha256:0bc3020d05f1e08b41f1c5d54650a157b1690cde7  5.1s
 => => sha256:612a5de913f35f1f4648862a6adc5847a92 4.20kB / 4.20kB  15.3s
 => => sha256:b86d81a99d4163dc88ac7c8a0a8d5bc8f 34.46MB / 34.46MB  22.7s
 => => sha256:5dd61d4ad9e84bebb71328a641516ac0ac0 2.29MB / 2.29MB  16.6s
 => => sha256:7aae823459655819b0bcb791763166b6a9643b7 282B / 282B  16.8s
 => => extracting sha256:a110e58716600c199fc95f633b30735c33a25b5ad  0.7s
 => => extracting sha256:83d3c0fa203acbade733bff627daa75b84c97f9d0  0.6s
 => => extracting sha256:a8fd09c11b021b756b7a92a4f70a3d444ce7e63a1  4.0s
 => => extracting sha256:14feb89c4a52c44f0a7e23a46ebb76e1c25c4b291  8.0s
 => => extracting sha256:612a5de913f35f1f4648862a6adc5847a922ae6a4  0.1s
 => => extracting sha256:b86d81a99d4163dc88ac7c8a0a8d5bc8fab559051  1.7s
 => => extracting sha256:5dd61d4ad9e84bebb71328a641516ac0ac01b1584  0.1s
 => => extracting sha256:7aae823459655819b0bcb791763166b6a9643b7b4  0.0s
 => [internal] load build context                                  20.7s
 => => transferring context: 19.39MB                               20.5s
 => [2/4] WORKDIR /app                                              1.2s
 => [3/4] COPY . .                                                  0.9s
 => ERROR [4/4] RUN NO_EMAIL_AUTOFILL=true node setup              93.3s
------
 > [4/4] RUN NO_EMAIL_AUTOFILL=true node setup:
#8 0.579 ▶️  Starting workshop setup...
#8 0.807       Running the following command: npx "https://gist.github.com/kentcdodds/bb452ffe53a5caa3600197e1d8005733" -q
#8 6.401     ▶️  Starting: System Validation
#8 6.401           Ensuring the correct versions of tools are installed on this computer.
#8 6.401           Running the following command: npx "https://gist.github.com/kentcdodds/abbc32701f78fa70298d444c2303b6d9"
#8 13.05 npm notice
#8 13.05 npm notice New minor version of npm available! 7.15.1 -> 7.19.1
#8 13.05 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.19.1>
#8 13.05 npm notice Run `npm install -g npm@7.19.1` to update!
#8 13.05 npm notice
#8 13.06     ✅  Success: System Validation
#8 13.06
#8 13.06
#8 13.06     ▶️  Starting: Dependency Installation
#8 13.06           Installing third party code dependencies so the workshop works properly on this computer.
#8 13.06           Running the following command: npm install --legacy-peer-deps --no-save
#8 92.84 npm ERR! code 1
#8 92.84 npm ERR! path /app/node_modules/husky
#8 92.84 npm ERR! command failed
#8 92.84 npm ERR! command sh -c node husky install
#8 92.84 npm ERR! /app/node_modules/please-upgrade-node/index.js:7
#8 92.84 npm ERR!   if (semverCompare(currentVersion, requiredVersion) === -1) {
#8 92.84 npm ERR!       ^
#8 92.84 npm ERR!
#8 92.84 npm ERR! TypeError: semverCompare is not a function
#8 92.84 npm ERR!     at pleaseUpgradeNode (/app/node_modules/please-upgrade-node/index.js:7:7)
#8 92.84 npm ERR!     at Object.<anonymous> (/app/node_modules/husky/husky.js:7:1)
#8 92.84 npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1109:14)
#8 92.84 npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
#8 92.84 npm ERR!     at Module.load (node:internal/modules/cjs/loader:989:32)
#8 92.84 npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:829:14)
#8 92.84 npm ERR!     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:76:12)
#8 92.84 npm ERR!     at node:internal/main/run_main_module:17:47
#8 92.92
#8 92.92 npm ERR! A complete log of this run can be found in:
#8 92.92 npm ERR!     /root/.npm/_logs/2021-07-01T21_35_33_822Z-debug.log
#8 92.94     🚨  Failure: Dependency Installation. Please review the messages above for information on how to troubleshoot and resolve this issue.
#8 92.95 npm notice
#8 92.95 npm notice New minor version of npm available! 7.15.1 -> 7.19.1
#8 92.95 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v7.19.1>
#8 92.95 npm notice Run `npm install -g npm@7.19.1` to update!
#8 92.95 npm notice
------
executor failed running [/bin/sh -c NO_EMAIL_AUTOFILL=true node setup]: exit code: 1
ERROR: Service 'node' failed to build : Build failed.

I'll try to debug in case I found the solution, it seems only related to my machine, I'm guessing no one else is having this problem.

**Comment by kentcdodds** [comment]
Yeah, this is very strange. I dug around the code and I have no idea why this could be happening. The package that's referenced in the error hasn't received any updates in years 🙃

Nobody else has reported anything like this I'm afraid 😬

You could remove `husky` from the `package.json` and delete the `package-lock.json`. I believe the project should still work.

**Comment by Jairanpo** [comment]
Hey!, I do get further, still getting an error but I have no clue yet:

npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm WARN deprecated rollup-plugin-babel@4.4.0: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-babel.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintained
npm WARN deprecated sane@4.1.0: some dependency vulnerabilities fixed, support for node < 10 dropped, and newer ECMAScript syntax/features added
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'

added 2626 packages, removed 1223 packages, changed 315 packages, and audited 3563 packages in 2m

213 packages are looking for funding
  run `npm fund` for details

26 vulnerabilities (8 moderate, 18 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
    ✅  Success: Dependency Installation


    ▶️  Starting: Project Validation
          Running validation checks to ensure dependencies were installed properly
          Running the following command: npm run validate -s
    🚨  Failure: Project Validation. Please review the messages above for information on how to troubleshoot and resolve this issue.

**Comment by Jairanpo** [comment]
D:\development\epicreact\react-fundamentals>npm run setup

> @kentcdodds/react-fundamentals@1.0.0 setup
> node setup

▶️  Starting workshop setup...
      Running the following command: npx "https://gist.github.com/kentcdodds/bb452ffe53a5caa3600197e1d8005733" -q
    ▶️  Starting: System Validation
          Ensuring the correct versions of tools are installed on this computer.
          Running the following command: npx "https://gist.github.com/kentcdodds/abbc32701f78fa70298d444c2303b6d9"
    ✅  Success: System Validation


    ▶️  Starting: Dependency Installation
          Installing third party code dependencies so the workshop works properly on this computer.
          Running the following command: npm install --legacy-peer-deps --no-save

up to date, audited 3563 packages in 4s

213 packages are looking for funding
  run `npm fund` for details

26 vulnerabilities (8 moderate, 18 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
    ✅  Success: Dependency Installation


    ▶️  Starting: Project Validation
          Running validation checks to ensure dependencies were installed properly
          Running the following command: npm run validate -s
    🚨  Failure: Project Validation. Please review the messages above for information on how to troubleshoot and resolve this issue.

**Comment by Jairanpo** [comment]
Ok, it seems that actually, my hard drive is having problems when dealing with the installation. I deleted, updated, and try a ton of possible ways to fix this problem (even docker), but it only works when I switch to my internal C: drive, I do have like 600GB of free space in that device but somehow just doesn't work, it is a thunderbolt 3 drive Rocket XTRM-Q, I have used it for some other projects with npm and yarn without problem but maybe something just isn't supported. Anyhow, I just share that in the remote case someone is having the same issue. Thank you very much @kentcdodds for your quick response! :)
